File: snippets/rust/src/list_payments.rs
use std::sync::Arc;

use anyhow::Result;
use breez_sdk_core::*;

async fn list_payments(sdk: Arc<BreezServices>) -> Result<Vec<Payment>> {
    // ANCHOR: list-payments
    let payments = sdk.list_payments(ListPaymentsRequest::default()).await?;
    // ANCHOR_END: list-payments

    Ok(payments)
}

async fn list_payments_filtered(sdk: Arc<BreezServices>) -> Result<Vec<Payment>> {
    // ANCHOR: list-payments-filtered
    let payments = sdk
        .list_payments(ListPaymentsRequest {
            filters: Some(vec![PaymentTypeFilter::Sent]),
            from_timestamp: Some(1696880000),
            include_failures: Some(true),
            ..Default::default()
        })
        .await?;
    // ANCHOR_END: list-payments-filtered

    Ok(payments)
}

File: snippets/go/list_payments.go
package example

import (
	"log"

	"github.com/breez/breez-sdk-go/breez_sdk"
)

func ListPayments() error {
	// ANCHOR: list-payments
	payments, err := sdk.ListPayments(breez_sdk.ListPaymentsRequest{})
	if err != nil {
		return err
	}
	log.Printf("%#v", payments)
	// ANCHOR_END: list-payments
	return nil
}

func ListPaymentsFiltered() error {
	// ANCHOR: list-payments-filtered
	filters := []breez_sdk.PaymentTypeFilter{breez_sdk.PaymentTypeFilterSent}
	fromTimestamp := int64(1696880000)
	includeFailures := true
	listPaymentsRequest := breez_sdk.ListPaymentsRequest{
		Filters:         &filters,
		FromTimestamp:   &fromTimestamp,
		IncludeFailures: &includeFailures,
	}
	payments, err := sdk.ListPayments(listPaymentsRequest)
	if err != nil {
		return err
	}
	log.Printf("%#v", payments)
	// ANCHOR_END: list-payments-filtered
	return nil
}
